build.sh

```diff
@@ -220,7 +220,7 @@ generate_event_logging_sources()
     fi
 
     echo "Laying out dynamically generated EventPipe Implementation"
-    $PYTHON -B $__PythonWarningFlags "$__ProjectRoot/src/scripts/genEventPipe.py" --man "$__ProjectRoot/src/vm/ClrEtwAll.man" --intermediate "$__OutputEventingDir/eventpipe"
+    $PYTHON -B $__PythonWarningFlags "$__ProjectRoot/src/scripts/genEventPipe.py" --man "$__ProjectRoot/src/vm/ClrEtwAll.man" --intermediate "$__OutputEventingDir/eventpipe" --nonextern
```

Comment:
From what I can remember, --noextern means is that the script doesn't place ```extern "C"``` before some of the methods in the build generated native runtime event functions.  This shouldn't actually change what gets built.  Can you tell me more about how this helps?

Reply:
@brianrob Yes, you right and this exactly why my change breaks all the Linux testing in this PR due to `InitProviderAndEvents` not being decorated with `extern "C"`

`--noextern` has also changes this during genEventPipe.py run
```
if extern: cmake_file.write("add_library")
else: cmake_file.write("add_library_clr")
```
The difference is that in `add_library_clr` we check if the library belongs to `CLR_CROSS_COMPONENTS_LIST` and we doesn't add the library to build targets if it doesn't.

Reply:
BTW, in Windows build.cmd we also run genEventPipe.py with --noextern - and my thought was to make Linux build.sh to be consistent with build.cmd

Reply:
@brianrob I wonder if need to generate all the eventing infrastructure in case of cross-arch components at all?

Reply:
Discussed with @brianrob offline the issue and the attempt now is to revert "noextern" change in build.sh and unconditionally use `add_library_clr(eventpipe` in genEventPipe.py and see if all the tracing tests pass